crd-generator/api/src/main/java/io/fabric8/crd/generator/CustomResourceInfo.java

```diff
@@ -148,4 +152,21 @@ public static CustomResourceInfo fromClass(Class<? extends CustomResource> custo
       throw KubernetesClientException.launderThrowable(e);
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    CustomResourceInfo that = (CustomResourceInfo) o;
+    return Objects.equals(crdName() + version, that.crdName() + that.version);
```

Comment:
This might be veeeery dangerous:

(crdName() == "val", version == "ue")
is equal to (crdName() == "va", version == "lue")

I think even the hashCode implementation might not follow the hashCode/equals contract

Reply:
This is basically `apiVersion` (in a simpler form). Aren't 2 CRs with the same `apiVersion` the same?

Reply:
OK, but I'm more worried about the hashCode part, should be replaced by ----> `Objects.hash(crdName() + version);`

Reply:
Also, you're not considering the `/` separator in apiVersion that would inevitably make `va/lue` different to `val/ue`

Reply:
Done.